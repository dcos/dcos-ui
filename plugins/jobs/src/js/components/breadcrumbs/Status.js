import PropTypes from "prop-types";
import React from "react";
import TaskStates from "#PLUGINS/services/src/js/constants/TaskStates";
import BreadcrumbSupplementalContent from "#SRC/js/components/BreadcrumbSupplementalContent";

export default function Status({ status }) {
  return (
    <BreadcrumbSupplementalContent>
      <div className="service-page-header-status muted">
        ({TaskStates[status].displayName})
      </div>
    </BreadcrumbSupplementalContent>
  );
}

Status.propTypes = {
  status: PropTypes.string.isRequired
};
