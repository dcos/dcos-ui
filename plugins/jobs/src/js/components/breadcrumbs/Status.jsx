import { Trans } from "@lingui/macro";
import PropTypes from "prop-types";
import React from "react";
import TaskStates from "#PLUGINS/services/src/js/constants/TaskStates";
import BreadcrumbSupplementalContent from "#SRC/js/components/BreadcrumbSupplementalContent";

export default function Status({ status }) {
  return (
    <BreadcrumbSupplementalContent>
      <div className="service-page-header-status muted">
        (<Trans render="span" id={TaskStates[status].displayName} />)
      </div>
    </BreadcrumbSupplementalContent>
  );
}

Status.propTypes = {
  status: PropTypes.string.isRequired
};
