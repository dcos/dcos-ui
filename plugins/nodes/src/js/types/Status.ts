import { i18nMark } from "@lingui/react";
import StatusIcon from "#SRC/js/constants/StatusIcon";
import Node from "#SRC/js/structs/Node";

const build = (
  priority: number,
  icon: StatusIcon,
  displayName: string
): Status => ({ priority, icon, displayName });

const active = build(2, StatusIcon.SUCCESS, i18nMark("Active"));
const deactivated = build(3, StatusIcon.STOPPED, i18nMark("Deactivated"));
const drained = build(5, StatusIcon.STOPPED, i18nMark("Drained"));
const draining = build(4, StatusIcon.LOADING, i18nMark("Draining"));
const unknown = build(1, StatusIcon.ERROR, i18nMark("Unknown"));

const fromNode = (n: Node) => {
  if (!n.isDeactivated()) {
    return active;
  }

  const info = n.getDrainInfo();
  if (info === undefined) {
    return deactivated;
  }

  switch (info.state) {
    case "DRAINING":
      return draining;
    case "DRAINED":
      return drained;
    case "UNKNOWN":
      return unknown;
  }
};

export type Status = {
  displayName: string;
  icon: StatusIcon;
  priority: number;
};

export const Status = {
  // The filters to be displayed in the NodesPage.
  // Should not be a concern of this module but feels still better than exporting
  // the statuses.
  filters: {
    active,
    deactivated,
    draining,
    drained
  },
  fromNode
};
