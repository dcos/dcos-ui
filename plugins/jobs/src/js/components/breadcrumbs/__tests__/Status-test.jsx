import React from "react";
import renderer from "react-test-renderer";

import Status from "../Status";

describe("Status", () => {
  // a subset of states for testing
  const states = [
    "TASK_RUNNING",
    "TASK_STARTING",
    "TASK_STARTED",
    "TASK_FAILED"
  ];
  for (const state of states) {
    it(`renders ${state}`, () => {
      expect(
        renderer.create(<Status status={state} />).toJSON()
      ).toMatchSnapshot();
    });
  }
});
