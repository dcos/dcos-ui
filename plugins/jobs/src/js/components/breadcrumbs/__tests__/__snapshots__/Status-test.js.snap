// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status renders TASK_FAILED 1`] = `
<div
  className="breadcrumb__content breadcrumb__content--supplemental"
>
  <div
    className="service-page-header-status muted"
  >
    (
    Failed
    )
  </div>
</div>
`;

exports[`Status renders TASK_RUNNING 1`] = `
<div
  className="breadcrumb__content breadcrumb__content--supplemental"
>
  <div
    className="service-page-header-status muted"
  >
    (
    Running
    )
  </div>
</div>
`;

exports[`Status renders TASK_STARTED 1`] = `
<div
  className="breadcrumb__content breadcrumb__content--supplemental"
>
  <div
    className="service-page-header-status muted"
  >
    (
    Started
    )
  </div>
</div>
`;

exports[`Status renders TASK_STARTING 1`] = `
<div
  className="breadcrumb__content breadcrumb__content--supplemental"
>
  <div
    className="service-page-header-status muted"
  >
    (
    Starting
    )
  </div>
</div>
`;
